webapp/lib/model.g.dart

```diff
@@ -447,3 +457,10 @@ abstract class DocBatchUpdate {
 
 // ======================================================================
 // Core pub/sub utilities
+
+/// A pub/sub based mechanism for updating documents
+abstract class DocPubSubUpdate {
+  /// Publish the given document changes,
+  /// where [changes] is a mapping of field name to new value
+  Future<bool> publishDocChange(String collectionName, List<String> docIds, Map<String, dynamic> changes);
```

Comment:
Can you delete fields with this? I think we'll need to be able to store nulls so we might need to seperate into 'changes' and 'deletes'?

Reply:
@lukechurch Good point. I think we'll need `setField` and `clearField` along with `addToField` and `removeFromField` for fields that are lists (e.g. tags). Will address in a subsequet PR.